LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,256 @@
+            {
+                VmSize = (AzureVmSize)Enum.Parse(typeof(AzureVmSize), settingsDeployment.VmSize),
+                GameCertificateReferences = null,
+                Ports = PortMapping(),
+                MultiplayerServerCountPerVm = settingsDeployment.MultiplayerServerCountPerVm,
+                RegionConfigurations = settingsDeployment.RegionConfigurations?.Select(x => new BuildRegionParams()
+                {
+                    Region = x.Region,
+                    MaxServers = x.MaxServers,
+                    StandbyServers = x.StandbyServers,
+                    MultiplayerServerCountPerVm = settingsDeployment.MultiplayerServerCountPerVm,
+                    VmSize = (AzureVmSize)Enum.Parse(typeof(AzureVmSize), settingsDeployment.VmSize)
+
+                }).ToList(),
+                BuildName = settingsDeployment.BuildName,
+                GameAssetReferences = settings.AssetDetails?.Select(x => new AssetReferenceParams()
+                {
+                    FileName = GetAssetFileNameFromPath(x.LocalFilePath),
+                }).ToList(),
+                StartMultiplayerServerCommand = settings.ProcessStartParameters.StartGameCommand,
+                OsPlatform = settingsDeployment.OSPlatform
+            };
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithProcessBasedServerResponse>> CreateBuildWithProcessBasedServer(CreateBuildWithProcessBasedServerRequest request)
+        {
+            Console.WriteLine($"Starting deployment {request.BuildName} for titleId, regions  {string.Join(", ", request.RegionConfigurations.Select(x => x.Region))}");
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithProcessBasedServerAsync(request);
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithManagedContainerResponse>> CreateBuildWithManagedContainer(CreateBuildWithManagedContainerRequest request)
+        {
+            Console.WriteLine($"Starting deployment {request.BuildName} for titleId, regions  {string.Join(", ", request.RegionConfigurations.Select(x => x.Region))}");
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithManagedContainerAsync(request);
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithCustomContainerResponse>> CreateBuildWithCustomContainer(CreateBuildWithCustomContainerRequest request)
+        {
+            Console.WriteLine($"Starting deployment {request.BuildName} for titleId, regions  {string.Join(", ", request.RegionConfigurations.Select(x => x.Region))}");
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithCustomContainerAsync(request);
+        }
+
+        public List<PlayFab.MultiplayerModels.Port> PortMapping()
+        {
+            List<PlayFab.MultiplayerModels.Port> ports = null;
+
+            foreach (var portList in settings.PortMappingsList)
+            {
+                ports = portList?.Select(x => new PlayFab.MultiplayerModels.Port()
```

Comment:
this call will overwrite the ports variable every time so that only the last portlist in the portmappinglists gets put in. Is that intentional?

If it's not, what you probably want to do is something like

var ports = new List<PlayFab.MultiplayerModels.Port>();

ports.AddRange( portList?.Select(...))

That way you'll always return an empty list (which is usually good) and you'll be adding all the ports that apply here.

Reply:
Wow....thank you for pointing this out @KalonJ-MSFT